src/git_repo_iter.rs

```diff
@@ -26,7 +26,7 @@ enum CallState {
 ///  2. Empty iterator: If we get to the end of a sub-iterator, pop & start from the top
 ///
 pub struct GitRepoIter<'a> {
-    config_iter: hash_map::Iter<'a, String, Rc<WatchConfig>>,
+    config_iter: btree_map::Iter<'a, String, Rc<WatchConfig>>,
```

Comment:
Its a shame that changes like this propagate so far. In other languages I'd use an interface, but that seems wrong here. What would be the equivalent in Rust? `Rc<dyn Iterator<_>>`?

Reply:
I'm not so sure on that one. I'm fairly sure `dyn` results in more memory management needing to be done at runtime and so a slightly higher memory footprint, so I'd probably advise keeping it as-is for now at least.